### ISSUE TITLE
Running flutter packages pub upgrade failed with the following output

### ISSUE THREAD

**Issue by pulyaevskiy** [dart-lang/pub#1809]
Package: https://pub.dartlang.org/packages/parallax_image#-analysis-tab-

```
ERR: Invalid argument(s): Minimum version ("2.0.0-dev.23.0") must be less than maximum ("2.0.0-dev.16.0").
ERR: package:pub_semver/src/version_range.dart 59         new VersionRange
 package:pub/src/pubspec.dart 366                     Pubspec._ensureEnvironment
 package:pub/src/pubspec.dart 331                     Pubspec.dartSdkConstraint
 package:pub/src/solver/backtracking_solver.dart 511  BacktrackingSolver._checkPubspecMatchesSdkConstraint
 package:pub/src/solver/backtracking_solver.dart 474  BacktrackingSolver._checkVersion
 ===== asynchronous gap ===========================
 package:pub/src/solver/backtracking_solver.dart 432  BacktrackingSolver._findValidVersion.<fn>
 ===== asynchronous gap ===========================
 dart:async                                           Future.doWhile
 package:pub/src/solver/backtracking_solver.dart 430  BacktrackingSolver._findValidVersion
 package:pub/src/solver/backtracking_solver.dart 337  BacktrackingSolver._versionQueueFor
 ===== asynchronous gap ===========================
 package:pub/src/solver/backtracking_solver.dart 290  BacktrackingSolver._solve.<fn>
 ===== asynchronous gap ===========================
 dart:async                                           _completeOnAsyncReturn
 package:pub/src/solver/backtracking_solver.dart 315  BacktrackingSolver._solve.<fn>
 dart:async                                           _completeOnAsyncReturn
 package:pub/src/solver/version_selection.dart 64     VersionSelection.select
 dart:async                                           _completeOnAsyncReturn
 package:pub/src/solver/version_selection.dart 91     VersionSelection._addDependencies
 dart:async                                           _completeOnAsyncReturn
 package:pub/src/solver/version_selection.dart 91     VersionSelection._addDependencies
ERR: This is an unexpected error.
```

I have:
```bash
$ flutter --version
Flutter 0.1.1-pre.24 • channel master • https://github.com/flutter/flutter.git
Framework • revision bfc6833459 (9 days ago) • 2018-02-12 14:10:39 -0800
Engine • revision f5a4a93787
Tools • Dart 2.0.0-dev.22.0
```

And `flutter packages pub upgrade` outputs:

```bash
Resolving dependencies...
Overriding the upper bound Dart SDK constraint to <=2.0.0-dev.22.0 for the following packages:

charcode, collection, flutter, http, http_parser, path, sky_engine, source_span, string_scanner, vector_math

To disable this you can set the PUB_ALLOW_PRERELEASE_SDK system environment variable to `false`, or you can silence this message by setting it to `quiet`.
  async 2.0.3 (2.0.4 available)
  charcode 1.1.1
  collection 1.14.5
  flutter 0.0.0 from sdk flutter
  http 0.11.3+16
  http_parser 3.1.1
  meta 1.1.1 (1.1.2 available)
  path 1.5.1
  sky_engine 0.0.99 from sdk flutter
  source_span 1.4.0
  string_scanner 1.0.2
  typed_data 1.1.4 (1.1.5 available)
  vector_math 2.0.5
No dependencies changed.
```

**Comment by nex3** [comment]
This is a bug that was fixed by #1782. Either your version of Flutter is out of date, or Flutter's version of pub is.

**Comment by pulyaevskiy** [comment]
According to https://github.com/flutter/flutter/issues/14771 it seems like Pub needs some help to fix this. I've upgraded to latest flutter dev channel and it works on my end.
